### PR TITLE
Make Harvester conditional.

### PR DESCRIPTION
**Pull request by MustaphaTR**
I'll probably need this for YR Slaves so free ones can't mine. Disabling the trait removes the resources stored in the vehicle.

Testcase makes having Radar Dome disable the Ore Miners.

### CODE REVIEW

**Comment by reaperrr** [comment]
> Harvester must override ConditionalTrait's Created method instead of implementing INotifyCreated directly

**Comment by MustaphaTR** [comment]
Updated.

File: OpenRA.Mods.Common/Traits/Harvester.cs
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Harvester.cs]: Are we sure this doesn't cause the activity to be queued twice when the trait starts enabled? (Once in `Created` and once here.)

Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/Harvester.cs]: I'm not. I'm not sure about this at the first place tbh. But if we want it to be triggered when enabled, one in the Created can go. Tho it was crashing when you leave the game when this was in a EndFrameTask and seems to be needed in Created as the comment says.

Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/Harvester.cs]: Yeah, i checked. It crashes with EndFrameTask and comment still applies without it. So not sure how to go with this.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Harvester.cs]: I would simply remove this from `TraitEnabled`.

**Comment by MustaphaTR** [comment]
Updated.

**Comment by MlemandPurrs** [comment]
oh great. this will also help them getting paused while EMP'ed.

File: OpenRA.Mods.Common/Activities/DeliverResources.cs
Comment (pchote) [inline OpenRA.Mods.Common/Activities/DeliverResources.cs]: I believe we should also be calling `INotifyHarvesterAction.MovementCancelled` here, so pull this into its own case that calls `Cancel`?

File: OpenRA.Mods.Common/Activities/HarvestResource.cs
Comment (pchote) [inline OpenRA.Mods.Common/Activities/HarvestResource.cs]: Likewise here.

**Comment by MustaphaTR** [comment]
Updated.

Also i just realised i didn't put the testcase on a seperate commit, so don't merge with it by mistake.

File: OpenRA.Mods.Common/Activities/DeliverResources.cs
Comment (pchote) [inline OpenRA.Mods.Common/Activities/DeliverResources.cs:47]: I haven't tested this, but was expecting these to be above the `IsCanceling` check so that it could exit on the same tick. Does having this here work as expected?

**Comment by MustaphaTR** [comment]
Updated.

**Review by pchote — DISMISSED** [review]
LGTM. Testcase will need to be removed before merging.

**Review by Smittytron — DISMISSED** [review]
LGTM, please remove testcase.

**Comment by MustaphaTR** [comment]
Removed the testcase.

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/eea74cf8da65fd2828fd6e178f2b2f331e168f62...ec8299c17e5d30a3ecd8ae8cc8255e922aaa06ad)

**Pull request merged**
